src/backend/distributed/sql/udfs/citus_shards/latest.sql

```diff
@@ -27,9 +27,8 @@ JOIN
    pg_dist_partition
 ON
    pg_dist_partition.logicalrelid = pg_dist_shard.logicalrelid
+WHERE 
+   pg_dist_placement.shardstate = 1
 ORDER BY
    pg_dist_shard.logicalrelid::text, shardid
 ;
-
-ALTER VIEW citus.citus_shards SET SCHEMA pg_catalog;
-GRANT SELECT ON pg_catalog.citus_shards TO public;
```

Comment:
I assume the grants stay in place, because we do CREATE OR REPLACE during the upgrade?

Reply:
umm I guess if the view is dropped then this won't have the same priviliges so we might need to grant them again here

Reply:
> When CREATE OR REPLACE VIEW is used on an existing view, only the view's defining SELECT rule is changed. Other view properties, including ownership, permissions, and non-SELECT rules, remain unchanged. You must own the view to replace it (this includes being a member of the owning role).

Reply:
Grant can be omitted.